src/service/impl/bookmarkServiceImpl.ts

```diff
@@ -0,0 +1,120 @@
+import { Article, dataSource, User } from "skiosa-orm";
+import { Arg, Authorized, Mutation, Resolver, Query } from "type-graphql";
+import { Repository } from "typeorm";
+import { CurrentUser } from "../../model/context";
+import { UserInfo } from "../../model/jwt";
+import { PaginationArg } from "../../model/paginationArg";
+import { paginate } from "../../util/paginate";
+import { BookmarkService } from "../bookmarkService";
+
+@Resolver()
+export class BookmarkServiceImpl implements BookmarkService {
+  @Mutation(() => Boolean)
+  @Authorized("realm:skiosa-user")
+  async addBookmark(@CurrentUser() currentUserInfo: UserInfo, @Arg("articleId") articleId: number): Promise<boolean> {
+    const userRepository = dataSource.getRepository(User);
+
+    const currentUser = await getCurrentUser(userRepository, currentUserInfo);
+
+    return dataSource
+      .getRepository(Article)
+      .findOneBy({
+        id: articleId,
+      })
+      .then((article) => {
+        if (article) {
+          currentUser.bookmarks?.push(article);
+          userRepository.manager.save(currentUser);
+          return true;
+        } else {
+          throw new Error("Article does not exist");
+        }
+      })
+      .catch(() => {
+        throw new Error("Article does not exist");
+      });
+  }
+
+  @Mutation(() => Boolean)
+  @Authorized("realm:skiosa-user")
+  async deleteBookmark(
+    @CurrentUser() currentUserInfo: UserInfo,
+    @Arg("articleId") articleId: number
+  ): Promise<boolean> {
+    const userRepository = dataSource.getRepository(User);
+
+    const currentUser = await getCurrentUser(userRepository, currentUserInfo);
```

Comment:
getCurrentUser is able to throw errors. await is problematic. Errors will be catched if using then